xml_converter/integration_tests/run_tests.py

```diff
@@ -166,17 +166,24 @@ def main() -> bool:
             if not re.match(args.filter, testcase.name):
                 continue
 
-        xml_output_dir_path = os.path.join(output_parent_dirpath, "xml", testcase.name)
-        proto_output_dir_path = os.path.join(output_parent_dirpath, "proto", testcase.name)
+        output_xml_paths: Optional[List[str]] = None
+        output_proto_paths: Optional[List[str]] = None
 
-        os.makedirs(xml_output_dir_path, exist_ok=True)
-        os.makedirs(proto_output_dir_path, exist_ok=True)
+        if testcase.expected_output_xml_path:
```

Comment:
Unless there are multiple expected truthy/falsey values expected here I would rather these be explicit checks for values. Otherwise you will very quickly create undefined behavior when something unexpected evaluates to true or false because it happens to be truthy or falsey when you did not expect it to ever be an option.